server/src/main/java/org/opensearch/rest/action/cat/RestShardsAction.java

```diff
@@ -107,25 +108,20 @@ protected void documentation(StringBuilder sb) {
     @Override
     public RestChannelConsumer doCatRequest(final RestRequest request, final NodeClient client) {
         final String[] indices = Strings.splitStringByCommaToArray(request.param("index"));
-        final ClusterStateRequest clusterStateRequest = new ClusterStateRequest();
-        clusterStateRequest.local(request.paramAsBoolean("local", clusterStateRequest.local()));
-        clusterStateRequest.clusterManagerNodeTimeout(
-            request.paramAsTime("cluster_manager_timeout", clusterStateRequest.clusterManagerNodeTimeout())
+        final CatShardsRequest shardsRequest = new CatShardsRequest();
+        shardsRequest.setLocal(request.paramAsBoolean("local", shardsRequest.local()));
+        shardsRequest.setClusterManagerNodeTimeout(
+            request.paramAsTime("cluster_manager_timeout", shardsRequest.clusterManagerNodeTimeout())
         );
-        parseDeprecatedMasterTimeoutParameter(clusterStateRequest, request, deprecationLogger, getName());
-        clusterStateRequest.clear().nodes(true).routingTable(true).indices(indices);
-        return channel -> client.admin().cluster().state(clusterStateRequest, new RestActionListener<ClusterStateResponse>(channel) {
+        shardsRequest.setCancelAfterTimeInterval(request.paramAsTime("cancel_after_time_interval", NO_TIMEOUT));
+        shardsRequest.setIndices(indices);
+        parseDeprecatedMasterTimeoutParameter(shardsRequest, request, deprecationLogger, getName());
+        return channel -> client.execute(CatShardsAction.INSTANCE, shardsRequest, new RestResponseListener<CatShardsResponse>(channel) {
             @Override
-            public void processResponse(final ClusterStateResponse clusterStateResponse) {
-                IndicesStatsRequest indicesStatsRequest = new IndicesStatsRequest();
-                indicesStatsRequest.all();
-                indicesStatsRequest.indices(indices);
-                client.admin().indices().stats(indicesStatsRequest, new RestResponseListener<IndicesStatsResponse>(channel) {
-                    @Override
-                    public RestResponse buildResponse(IndicesStatsResponse indicesStatsResponse) throws Exception {
-                        return RestTable.buildResponse(buildTable(request, clusterStateResponse, indicesStatsResponse), channel);
-                    }
-                });
+            public RestResponse buildResponse(CatShardsResponse catShardsResponse) throws Exception {
```

Comment:
Can we verify as to what will be the status code on task cancellation exception ? Also would user facing message be something like this ?

```
      {
          "type": "task_cancelled_exception",
          "reason": "cancelled task with reason: elapsed time exceeded [1.1s >= 300ms]"
      }
```

Reply:
Error message will be like - 
```
curl -X GET "localhost:9200/_cat/shards?cancel_after_time_interval=2s&pretty"
{
  "error" : {
    "root_cause" : [
      {
        "type" : "task_cancelled_exception",
        "reason" : "The parent task was cancelled, shouldn't start any child tasks, Cancellation timeout of 2s is expired"
      }
    ],
    "type" : "task_cancelled_exception",
    "reason" : "The parent task was cancelled, shouldn't start any child tasks, Cancellation timeout of 2s is expired"
  },
  "status" : 500
}
```

Reply:
It will be like 
```
{
  "error" : {
    "root_cause" : [
      {
        "type" : "task_cancelled_exception",
        "reason" : "Cancellation timeout of 1s is expired"
      }
    ],
    "type" : "task_cancelled_exception",
    "reason" : "Cancellation timeout of 1s is expired"
  },
  "status" : 500
}
```